concise-mid-tag.cddl

```diff
@@ -74,22 +74,26 @@ identity-triple-record = [
   [ + verification-key-map ] ; DevID, or semantically equivalent
 ]
 
-pkix-base64-key-type = tstr
-pkix-base64-cert-type = tstr
+$rawkey-type-choice /= tagged-pkix-base64-key-type
+
+; Raw key in DER format base64-encoded.
+tagged-pkix-base64-key-type = #6.554(tstr)
+
+$authority-path-type-choice /= tagged-pkix-base64-cert-type
+
+; X.509 certificate chain in DER format base64-encoded
+tagged-pkix-base64-cert-type = #6.555(tstr)
 
 verification-key-map = {
-  ; Verification key in DER format base64-encoded.
-  ; Typically, but not necessarily a public key.
-  comid.key => pkix-base64-key-type
-  ; Optional X.509 certificate chain corresponding to the public key
-  ; in comid.key, encoded as an array of one or more base64-encoded
-  ; DER PKIX certificates.  The certificate containing the public key
-  ; in comid.key MUST be the first certificate.  This MAY be followed
-  ; by additional certificates, with each subsequent certificate
-  ; being the one used to certify the previous one.
-  ? comid.keychain => [ + pkix-base64-cert-type ]
+  raw-key-or-authority-path
```

Comment:
But this thread is leading away from the objective of moving forward with the endorsement spec. I think its OK to keep the basic structure of the verification-key-map with two  elements - a raw key and an array of authority-key that are optional but bounded by non-empty<> macro. This will have the least impact on the spec and still achieve the intended goal.

Reply:
Yeah, I thought about that, and got lazy when I realised the amount of churn that'd mean for my implementation :-)

Reply:
Not sure I understand your proposal.  Could you provide a CDDL sketch?

Reply:
Option 1:
```
identity-triple-record = [
  environment-map ; device identifier (instance or class)
  [ + $key-type-choice ]
```
In option 1, the $key-type-choice defines raw key, authority-key and collections of authority-key structures.

Option 2:
```
verification-key-map = {
    $key-type-choice 
}
```
Option 2 is the same as option 1 in terms of the various key structures that could be defined. Retains the map structure but that doesn't seem to add much since the map structure isn't extensible and only has one element.

Option 3:
```
verification-key-map = {
    ? comid.raw-key => $raw-key-type-choice 
    ? comid.authority-key => $authority-key-type-choice
}
```
In option 3 $raw-key-type-choice defines only raw key formats and $authority-key-type-choice only defines keys with some form of authority component including signature, certificate, COSE_Sign, plus it includes sets of authority-key objects such as [ $authority-key-type-choice ] if that doesn't already exist as part of the authority structure.

All options push much of the complexity of how to represent keys into a tagged type structure that is extensible and the definition is decoupled from the main schema.